logstash-core/lib/logstash/logging/slow_logger.rb

```diff
@@ -0,0 +1,67 @@
+# encoding: utf-8
+require "logstash/namespace"
+require "logstash/logging/slow_logger/freq_items"
+
+# This module contains necessary classes to handle slow log
+# operations.
+module LogStash; module Logging
+
+  # Slow log null logger, this logger does
+  # nothing when used.
+  class NullLogger
+    def log(event, threshold, time, data={})
+    end
+  end
+
+  # Generic slow logger, this class is responsible of
+  # reporting to the specific logger and holding an
+  # aggregated view of it's reported events.
+  class SlowLogger
+
+    attr_reader :logger, :freq_items, :settings
+
+    def initialize(name="loggers.slow", settings=LogStash::SETTINGS, params={})
+      @logger     = LogStash::Logging::Logger.new(name)
+      @freq_items = LogStash::Logging::Util::FreqItems.new
+      @settings   = settings
+    end
+
+    def log(event, threshold, took_in_seconds, data={})
+      max_time = setting(threshold).to_i
+      return if max_time == 0 || took_in_seconds <= max_time
+
+      level, operation_namespace = split_fields(threshold)
+      if setting("slowlog.plugins.context")
```

Comment:
Can we call this `slowlog.enable.source` or `slowlog.log.source` or `slowlog.log.event`? There are other context we are logging, which we don't control using this setting.